client/src/components/Tea/TeaCollectionTable/TeaCollectionTable.jsx

```diff
@@ -0,0 +1,155 @@
+              </div>
+              <div className="control">
+                <button type="submit" className="button is-small">
+                  Filter
+                </button>
+              </div>
+            </div>
+          </form>
+          {this.props.filtered ? (
+            <div className="control">
+              <button
+                className="button is-primary is-small"
+                onClick={this.props.handleClearFilterClick}
+              >
+                Clear Filter
+              </button>
+            </div>
+          ) : null}
+        </div>
+        <table className="table is-striped is-fullwidth">
+          <thead>
+            <tr>
+              {this.props.columnHeaders.map(colHeaderObj => {
+                return (
+                  <th key={colHeaderObj.colName}>
+                    {colHeaderObj.colTitle + " "}
+                    <button
+                      className="button is-small"
+                      type="button"
+                      aria-pressed="false"
+                      onClick={() =>
+                        this.props.handleSortClick(
+                          colHeaderObj.colName,
+                          this.props.sortColumnHandler(colHeaderObj.colName)
+                            ? "desc"
+                            : "asc"
+                        )
+                      }
+                    >
+                      <span className="icon">
+                        <i className="fas">
+                          <FontAwesomeIcon
+                            icon={
+                              this.props.sortColumnHandler(colHeaderObj.colName)
+                                ? "angle-up"
+                                : "angle-down"
+                            }
+                          />
+                        </i>
+                      </span>
+                    </button>
+                  </th>
+                );
+              })}
+              <th />
+              <th />
+            </tr>
+          </thead>
+          <tbody>
+            {this.props.teaIDs.map(teaID => {
+              const tea = this.props.allTeas[teaID];
+              return (
+                <tr key={tea.id}>
+                  <td>
+                    <Link to={"tea/" + tea.id}>{tea.name}</Link>
+                  </td>
+                  <td>{tea.brand}</td>
+                  <td>{tea.teaType}</td>
+                  <td>{tea.servings}</td>
+                  <td>
+                    <Link to={"/update-tea/" + tea.id}>Edit</Link>
+                  </td>
+                  <td>
+                    <button
+                      className="button is-danger is-small"
+                      onClick={() => this.props.handleDeleteClick(tea.id)}
+                    >
+                      X
+                    </button>
+                  </td>
+                </tr>
+              );
+            })}
+            {this.props.teaIDs.length === 0 ? (
```

Comment:
I think @JeffreyATW has made this comment elsewhere, but you can typically write these sorts of conditionals as AND statements rather than ternaries (e.g. `this.props.teaIDs.length === 0 && {YOUR CODE HERE}`), eliminating the need to handle the false condition (the whole {YOUR CODE HERE} becomes a no-op when `this.props.teaIDs.length === 0` evaluates to false).

Reply:
You can get a good overview of all of this here: https://reactjs.org/docs/conditional-rendering.html.  The technique we're talking about is https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator